cookbooks/ros2_windows/recipes/xmllint.rb

```diff
@@ -1,18 +1,19 @@
+xmllint_url = node['ros2_windows']['xmllint']['xmllint_download_url']
 seven_zip_archive 'libxml2' do
   path 'C:\\xmllint'
-  source 'https://www.zlatkovic.com/pub/libxml/64bit/libxml2-2.9.3-win32-x86_64.7z'
+  source xmllint_url + 'libxml2-2.9.3-win32-x86_64.7z'
```

Comment:
This is a minor nit, but using `+` here seems a little fragile.  If someone forgets to add the `/` to the end of the download_url, this will produce an invalid URL.  Does ruby/chef have the equivalent of `'/'.join([a, b])` that Python has?

Reply:
It does
```suggestion
  source [xmllint_url, 'libxml2-2.9.3-win32-x86_64.7z'].join('/')
```

and that would have the much less fatal effect of doubling the '/' if there is one present.
From a "keep it as obvious as possible" perspective I would probably opt for adding a leading slash to the file name and maintaining the concatenation rather than using a join. It has the additional, completely pointless, advantage of eliminating an additional array and string allocation (the array enclosing the url prefix and filename, and the literal `'/'`). 

```suggestion
  source xmllint_url + '/libxml2-2.9.3-win32-x86_64.7z'
```